processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -77,6 +78,7 @@
     DECORATOR_CONSTRUCTOR( "Specified decorator type has no default constructor nor a constructor with a single parameter accepting the decorated mapper type." ),
 
     GENERAL_NO_IMPLEMENTATION( "No implementation type is registered for return type %s." ),
+    GENERAL_ABSTRACT_RETURN_TYPE( "The return type %s is abstract class or interface. Provide non abstract/non interface result type or a factory method." ),
```

Comment:
`... is an abstarct class or interface. Provide a non abstract / non interface result type .. `?
(adding "an", "a", and spaces around "/")